kore/src/Kore/Attribute/SourceLocation.hs

```diff
@@ -0,0 +1,78 @@
+{-|
+Module      : Kore.Attribute.SourceLocation
+Description : Source and location attribute
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+Maintainer  : vladimir.ciobanu@runtimeverification.com
+
+-}
+module Kore.Attribute.SourceLocation
+    ( SourceLocation (..)
+    , Source (..)
+    , Location (..)
+    ) where
+
+import           Control.Applicative
+                 ( (<|>) )
+import           Control.DeepSeq
+                 ( NFData )
+import           Data.Default
+import           Data.Text.Prettyprint.Doc
+                 ( Pretty )
+import qualified Data.Text.Prettyprint.Doc as Pretty
+import           GHC.Generics
+                 ( Generic )
+
+import           Kore.Attribute.Location
+                 ( LineColumn (..), Location (..) )
+import qualified Kore.Attribute.Location as Location
+import           Kore.Attribute.Parser
+                 ( ParseAttributes (..) )
+import           Kore.Attribute.Source
+                 ( Source (..) )
+import qualified Kore.Attribute.Source as Source
+
+data SourceLocation = SourceLocation
+    { location :: !(Maybe Location)
+    , source   :: !(Maybe Source)
+    } deriving (Eq, Ord, Show, Generic)
+
+instance NFData SourceLocation
+
+instance Default SourceLocation where
+    def = SourceLocation Nothing Nothing
+
+instance ParseAttributes SourceLocation where
+    parseAttribute patt SourceLocation { location, source } = do
```

Comment:
Other "aggregate" attributes define their parser as
```.hs
    lensLocation (parseAttribute patt)
    >=> lensSource (parseAttribute patt)
```
which handles duplicate attributes correctly. Could we do that here, too?